crosspoint_reader/driver.py

```
@@ -293,47 +330,66 @@ def upload_books(self, files, names, on_card=None, end_session=True, metadata=No
+                    target_dir = self._ensure_dir(base_path, subdirs)
+                else:
+                    target_dir = base_path
+
+                if target_dir == '/':
+                    lpath = '/' + filename
+                else:
+                    lpath = target_dir + '/' + filename
+
+                def _progress(sent, size):
+                    if size > 0:
+                        self.report_progress((i + sent / float(size)) / float(total),
```

Ruff (0.15.2)

[warning] Function definition does not bind loop variable `i`

(B023)